src/net_processing.h

```diff
@@ -47,8 +47,12 @@ class PeerManager : public CValidationInterface, public NetEventsInterface
     /** Whether this node ignores txs received over p2p. */
     virtual bool IgnoresIncomingTxs() = 0;
 
-    /** Relay transaction to all peers. */
-    virtual void RelayTransaction(const uint256& txid, const uint256& wtxid)
+    /**
+     * Relay transaction to every node.
+     * The announcement protocol (flooding or reconciliation) will be chosen based on from which
+     * peer the transaction came to us. For transactions originated locally, pass from=nullopt.
+     */
+    virtual void RelayTransaction(const uint256& txid, const uint256& wtxid, std::optional<NodeId> from=std::nullopt)
```

Comment:
There's no need to update this public interface method. The only users of this method outside net_processing will _always_ call `RelayTransaction()` without a `from` parameter.

(it's also unnecessary to wrap an optional parameter in a `std::optional` since the fact that it's an optional parameter already gives you a way to indicate no value)

Reply:
Sorry I don't understand this comment, despite your elaborate explanation. Probably my lack of C++ expertise.

If I remove just this change in the `virtual void` line, I get a compilation error.

Reply:
Yes, you'll need to change the function signature in `PeerManagerImpl()` as well. An `override` function must have the same signature as the function it's overriding.